src/App.js

```diff
@@ -118,53 +178,263 @@ class App extends Component {
 
     players[player.name] = updatedPlayer;
 
-
     this.setState({
       players: players
     })
   }
 
+  setRotation = (type) => {
+    this.setState({ rotationType: type })
+  }
+
   render() {
-    const self = this;
+    
     return (
       <div>
         <button onClick={this.dealDeck}>Deal</button>
         <button onClick={this.playHand}>Play Hand</button>
         <div className="game-wrapper">
-          {_.map(this.state.players, function (player, idx) {
-            return (
-              <div className={"player-wrapper " + (idx) } key={idx}>
-                <div className={player.currentPlayer
-                  ? "current-player player-name"
-                  : "player-name"}>{player.name}</div>
-                <div className="player-stage">
-                  {player.hand && player.hand.map(function (card, idx) {
-                  return (
-                    <div key={idx}
-                      className={player.selectedCards.has(card) ? "selected card-box" : "card-box"}
-                      onClick={() => self.selectCard(player, card)}>
-                      {card.value} {card.suit} <br/>
-                      rank: {card.rank}
-                    </div>
-                  )
-                  })}
-                </div>
-              </div>
-            )
-          })}
-          <div className="player-stage stage">
-            {this.state.stage.map(function (card, idx) {
-              return(
-                <div className="card-box" key={idx}>
-                    {card.value} {card.suit}
-                </div>
-              )
-            })}
-          </div>
+          {
+            _.map(
+              this.state.players, 
+              (player, idx) => <Player player={player} actions={this.actions.players} idx={idx}/>)
+          }
+          <Stage stage={this.state.stage} rotationType={this.state.rotationType} actions={this.actions.stage}/>
         </div>
       </div>
     );
   }
 }
 
+let rotations = {
+  single: {
+    description: "You must play a single card of a higher rank than the card on the stage",
+    rules: [
+      (cards) => cards.length === 1,
+      (cards, stage) => {
+        return cards.length === stage.length
+      },
+      (cards, stage) => {
+        let _cards = Array.from(cards),
+            _stage = Array.from(stage);
+
+        return _cards.some(card => _stage.some(scard => card.rank > scard.rank));
+      }
+    ]
+  },
+
```

Comment:
single, pair, triple repeat a lot of the same rules. only difference is the length. Could this be a single rule that you pass the length to so that way we're not repeating ourselves.